### PR TITLE
fix(workflow): Fix permissions for editing Metric Alert Rules

### PR DESCRIPTION
**Pull request by billyvg**
The UI was never updated to use the "project" scoped endpoint for metric alert rules. The UI also locked the form to require `org:write` access, instead of `project:write`.

Needs https://github.com/getsentry/sentry/pull/16408 and https://github.com/getsentry/sentry/pull/16151

### CODE REVIEW

**Review by iProgramStuff — COMMENTED** [review]
This needs to come with backend changes I think. The project scoped API probably isn't using the new combined rule serializer.

Did you see that when you were developing this?

**Comment by billyvg** [comment]
@iProgramStuff yeah I was gonna wait for your PR to be merged first

File: src/sentry/incidents/endpoints/project_alert_rule_index.py
Comment (iProgramStuff) [inline src/sentry/incidents/endpoints/project_alert_rule_index.py]: This refactor (https://github.com/getsentry/sentry/pull/16546) removes the UnifiedAlertRuleSerializer

Comment (iProgramStuff) [reply src/sentry/incidents/endpoints/project_alert_rule_index.py]: once it's merged, we can leave this as-is (using AlertRuleSerializer)

**Review by iProgramStuff — CHANGES_REQUESTED** [review]

**Review by iProgramStuff — APPROVED** [review]

**Pull request merged**
